### ISSUE TITLE
Support https/tls for weed filer/mount

### ISSUE THREAD

**Issue by isjo-org** [seaweedfs/seaweedfs#4835]
Hi there!

We are already using seaweedfs with all combinations(5 volume server, 3 master, 3 filer, mount/webdav/s3) in our dev.env, so far, so happy!

Now we want to go to production with master/volume/filer/s3/mount without any private network between those servers, so only public ip + self signed cert

**Is there any progess for supporting TLS?**
Im wondering, if we are the only ones with this problem, as for now i didnt see any tickets for this problem?

for sure, i could use JWT to trust client/server, but content is still unencrypted on read/write :/

```
# volume server https options
# Note: work in progress!
#     this does not work with other clients, e.g., "weed filer|mount" etc, yet.
[https.client]
enabled = true
[https.volume]
cert = ""
key = ""
```

TIA!

br

**Comment by kmlebedev** [comment]
Hello, yes, encrypting traffic on volumes seems redundant, so there are few such configurations.

**Comment by NiceRath** [comment]
Greetings!

We would like to use SeaweedFS on distributed servers that communicate over public WAN.
So enrypting any kind of traffic is necessary!

Greetings
- Rath

**Comment by kmlebedev** [comment]
I just launched volume in tls mode, no problems with that.

**Comment by NiceRath** [comment]
Greetings!

My college @isjo-org may be able to provide the exact error messages next week, but the problems ocurr when actually using it. (and blocking any traffic to non-tls ports)
We currently are trying to get it working by replacing all the hardcoded references to 'http://' with a config-switch.
(to see references - just run `cd  seaweedfs/weed && grep -r '\"http://'`)

**Comment by NiceRath** [comment]
@kmlebedev Greetings!

May I ask if you have found the same issues as we have, while testing?

We have also sniffed the traffic using `tcpdump` and have found plaintext HTTP calls, even if TLS is enabled by config.

In regards to our 'PR' - we just wanted to show how we would approach the issue. But it is far from being clean, as we do not have an overview of the codebase..

**Comment by oivindoh** [comment]
> I just launched volume in tls mode, no problems with that.

Yes, it will listen with https, but e.g. weed mount is hardcoded to use http, which will error out since it'll be trying to make plaintext calls to a https service. I'm not sure @chrislusf is aware of this state at the moment - we might try to submit a new PR on this, but so far not familiar enough with the code base to make an honest attempt

**Comment by chrislusf** [comment]
https://github.com/seaweedfs/seaweedfs/pull/5766 by @vadimartynov has changed many places. 

@oivindoh You may want to continue from there.

**Comment by vadimartynov** [comment]
I tried to add TLS where possible in https://github.com/seaweedfs/seaweedfs/pull/5766
Try running weed `-config_dir="<path/to/the/folder/with/security.toml and other .toml configs>" mount`.
It might already be working; the config just might not be loading properly.

**Comment by oivindoh** [comment]
I think maybe places like [weed_write](https://github.com/seaweedfs/seaweedfs/blob/edef48533361dee2ae598782b85e233cc8110e50/weed/mount/weedfs_write.go#L41) is the issue referenced by places like [csi driver security config](https://github.com/seaweedfs/seaweedfs-csi-driver/blob/c19bcea8bd0465d6d9c0605980981ec8429d5faf/deploy/helm/seaweedfs-csi-driver/templates/configmap-security.yaml#L55)?
